### PR TITLE
Allow ClipRRect.borderRadius to support BorderRadiusDirectional

### LINKED ISSUE #91997: [Proposal] Allow `ClipRRect.borderRadius` to support `BorderRadiusDirectional`
Currently the `borderRadius` of `ClipRRect` is of type `BorderRadius`, 
So it is not possible to use a `ClipRRect` with `BorderRadiusDirectional`.

# Proposal
Allow `ClipRRect.borderRadius` to support `BorderRadiusDirectional`

### PR DESCRIPTION
**Pull request by thecalamiity**
Fixes #91997

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/clip_test.dart [hunk 1]
```diff
@@ -883,4 +883,90 @@ void main() {
...
+        child: const ClipRRect(
+          borderRadius: BorderRadiusDirectional.horizontal(
+            start: startRadius,
+            end: endRadius,
+          ),
+        ),
+      );
+    }
+
+    for(final TextDirection textDirection in TextDirection.values) {
```
Comment (nt4f04uNd) [inline packages/flutter/test/widgets/clip_test.dart:903]: space after for

File: packages/flutter/test/widgets/clip_test.dart [hunk 2]
```diff
@@ -883,4 +883,90 @@ void main() {
...
+                  },
+                ),
+              ),
+            ),
+          );
+        },
+      );
+    }
+
+    for(final TextDirection textDirection in TextDirection.values) {
```
Comment (nt4f04uNd) [inline packages/flutter/test/widgets/clip_test.dart:958]: space after for

**Review by nt4f04uNd — APPROVED** [review]
LGTM with those two spaces
This will need a secondary review

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
I restarted the failed check. If it doesn't pass, you may have to rebase this PR to the latest master.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by MrBirb** [comment]
I pushed an empty commit. Hopefully all tests will pass now

**Pull request merged**